src/models/hooks/activityReport.js

```diff
@@ -561,16 +567,54 @@ const beforeCreate = async (instance) => {
   copyStatus(instance);
 };
 
+const getActivityReportDocument = async (sequelize, instance) => {
+  const data = await collectModelData(
+    [instance.id],
+    AWS_ELASTIC_SEARCH_INDEXES.ACTIVITY_REPORTS,
+    sequelize,
+  );
+  return formatModelForAwsElasticsearch(
+    AWS_ELASTIC_SEARCH_INDEXES.ACTIVITY_REPORTS,
+    { ...data, ar: { ...instance.dataValues } },
+  );
+};
+
 const beforeUpdate = async (instance) => {
   copyStatus(instance);
 };
+
+const updateAwsElasticsearchIndexes = async (sequelize, instance) => {
+  // AWS Elasticsearch: Determine if we queue delete or update index document.
+  const changed = instance.changed();
+  if (Array.isArray(changed) && changed.includes('calculatedStatus')) {
+    if (instance.previous('calculatedStatus') !== REPORT_STATUSES.DELETED
+        && instance.calculatedStatus === REPORT_STATUSES.DELETED) {
+      // Delete Index Document for AWS Elasticsearch.
+      await scheduleDeleteIndexDocumentJob(
+        instance.id,
+        AWS_ELASTIC_SEARCH_INDEXES.ACTIVITY_REPORTS,
+      );
+    } else if (instance.previous('calculatedStatus') !== REPORT_STATUSES.SUBMITTED
```

Comment:
I assume this is "submitted" and not "approved" on purpose?

Reply:
I only ask because the UI that was proposed was a filter like all the others, and all the pages with filters expect "approved" reports only

Reply:
We can do it on any update if you think it would be better.